packages/ember-metal/lib/merge.js

```diff
@@ -16,6 +18,8 @@
   @public
 */
 export default function merge(original, updates) {
+  deprecate('Usage of `Ember.merge` is deprecated, use `Ember.assign` instead.', false, { id: 'ember-metal.merge', until: '3.0.0' });
```

Comment:
This will need to be flagged (so that we do not deprecate if the new feature flag is disabled).